### DISCUSSION TITLE
Few questions about the system.

### DISCUSSION THREAD

**Discussion by NoAce4U** [Vdauphin/HeartsAndMinds#1141] [Question and answer]
I see the restriction options and can somewhat grasp the idea of how they work. My question is, I'm having issues figuring out how exactly it calls in the cfg tree to do role based restrictions. If someone could point me in the right direction there. 

I'm also a bit confused on where it calls for default loadouts on spawns.

**Comment by Vdauphin** [comment]
Hello,
Restriction about what?
Cheers

**Reply by NoAce4U** [reply]
How do role based restrictions work... How do you modify them to work with only specific class names, not just the cfg tree. 

And how do you modify default loadouts. (On Spawn Loadouts)

**Reply by NoAce4U** [reply]
I'm also curious if there is a way to modify this to mot be DLC dependent. I know we could disable the two responsible for DLC dependency, but I'd like to keep them and possibly use other items in there place?

**Comment by Vdauphin** [comment]
> How do role based restrictions work... How do you modify them to work with only specific class names, not just the cfg tree.

here is the function you are looking for: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/arsenal/trait.sqf

> And how do you modify default loadouts. (On Spawn Loadouts)

Here you can change weapon uniform...
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L38-L62
Hhre is the function generating the loadout based on type of unit, the time, map etc:
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/arsenal/loadout.sqf

> I'm also curious if there is a way to modify this to mot be DLC dependent. 
There are no mod dependency

> but I'd like to keep them and possibly use other items in there place?

For chemical look here may be: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/chem/damage.sqf#L47-L55

Here for spectrum device: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/spect/updateDevice.sqf#L33

Cheers
